### PR TITLE
add new top variables

### PR DESCRIPTION
**Pull request by semrat**
I added the new top tag variables which are for: 
- calculate DeltaR between 2 tops
- get vectors of the top mass & eta & pT in the event
- update the best top mass part & add best top pT

### CODE REVIEW

File: Framework/include/RunTopTagger.h [hunk 1]
```diff
@@ -167,21 +167,54 @@ class RunTopTagger
...
+        {
+            topsEta.push_back(t->p().Eta());
+        }
+	
+	auto& topsPt = tr.createDerivedVec<double>("topsPt"+myVarSuffix_);
+        for(const auto* t : tops)
+        {
+            topsPt.push_back(t->p().Pt());
+        }
+
```
Comment (cmadrid1) [inline Framework/include/RunTopTagger.h]: Please don't loop through "tops" anymore than you have to. You can put all three for loops above into one.

Comment (semrat) [reply Framework/include/RunTopTagger.h]: OK. I updated and will push them back to gitHub.

File: Framework/include/RunTopTagger.h [hunk 2]
```diff
@@ -167,21 +167,47 @@ class RunTopTagger
...
+        if(ntops_ == 2)
+        {
+            dR_top1_top2 = tops.at(0)->p().DeltaR(tops.at(1)->p()); 
+        }
+
+	// -------------------------------------------------------------
+        // -- get vectors of the top mass & eta & pT & LV in the event
+        // -------------------------------------------------------------
+        auto& topsMass = tr.createDerivedVec<double>("topsMass"+myVarSuffix_);
+        auto& topsEta  = tr.createDerivedVec<double>("topsEta"+myVarSuffix_);
```
Comment (cmadrid1) [inline Framework/include/RunTopTagger.h]: Spacing here look broken

File: Framework/include/RunTopTagger.h [hunk 3]
```diff
@@ -167,21 +167,47 @@ class RunTopTagger
         // Get reconstructed tops and derive needed variables                            
         const std::vector<TopObject*>& tops = ttr.getTops();
         countTops(tops);
+	
+	// -------------------------------------
```
Comment (cmadrid1) [inline Framework/include/RunTopTagger.h]: Spacing here look broken

File: Framework/include/RunTopTagger.h [hunk 4]
```diff
@@ -167,21 +167,47 @@ class RunTopTagger
...
+        // -- Calculate DeltaR between 2 tops
+        // -------------------------------------
+	double dR_top1_top2 = -1;
+        if(ntops_ == 2)
+        {
+            dR_top1_top2 = tops.at(0)->p().DeltaR(tops.at(1)->p()); 
+        }
+
+	// -------------------------------------------------------------
```
Comment (cmadrid1) [inline Framework/include/RunTopTagger.h]: Spacing here look broken

File: Framework/include/RunTopTagger.h [hunk 5]
```diff
@@ -167,21 +167,47 @@ class RunTopTagger
...
+            topsMass.push_back(t->p().M());
+	    topsEta.push_back(t->p().Eta());
+	    topsPt.push_back(t->p().Pt());
+	    topsLV.push_back(t->p());
+        }
 
         const auto& candidateTops = ttr.getTopCandidates();
         double bestTopMass = -9999.9;
         double bestTopEta = -9999.9;
+	double bestTopPt = -9999.9;
```
Comment (cmadrid1) [inline Framework/include/RunTopTagger.h]: Spacing here look broken

File: Framework/include/RunTopTagger.h [hunk 6]
```diff
@@ -167,21 +167,47 @@ class RunTopTagger
...
         for(int iTop = 0; iTop < candidateTops.size(); ++iTop)
         {
             auto& top = candidateTops[iTop];
 
-            if(fabs(top.p().M() - 173.5) < fabs(bestTopMass - 173.5) && top.getNConstituents() == 3)
+            if(fabs(top.p().M() - 173.21) < fabs(bestTopMass - 173.21))
             {
                 bestTopMass = top.p().M();
                 bestTopEta = top.p().Eta();
+		bestTopPt = top.p().Pt();
```
Comment (cmadrid1) [inline Framework/include/RunTopTagger.h]: Spacing here look broken

File: Framework/include/RunTopTagger.h [hunk 7]
```diff
@@ -233,10 +259,13 @@ class RunTopTagger
         tr.registerDerivedVar("ntops_2jet"+myVarSuffix_, ntops_2jet_);
         tr.registerDerivedVar("ntops_1jet"+myVarSuffix_, ntops_1jet_);
         tr.registerDerivedVar("bestTopMassLV"+myVarSuffix_, bestTopMassLV?(bestTopMassLV->p()):(TLorentzVector()));
+	tr.registerDerivedVar("dR_top1_top2"+myVarSuffix_,dR_top1_top2);
```
Comment (cmadrid1) [inline Framework/include/RunTopTagger.h]: Spacing here look broken

**Pull request merged**